encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dql/EncryptMergedResult.java

```diff
@@ -44,15 +44,15 @@
     public boolean next() throws SQLException {
         return mergedResult.next();
     }
-    
+
     @Override
     public Object getValue(final int columnIndex, final Class<?> type) throws SQLException {
-        Object value = mergedResult.getValue(columnIndex, type);
-        if (null == value || !queryWithCipherColumn) {
-            return value;
+        Optional<Encryptor> encryptor;
+        if (!queryWithCipherColumn || !(encryptor = metaData.findEncryptor(columnIndex)).isPresent()) {
+            return mergedResult.getValue(columnIndex, type);
         }
-        Optional<Encryptor> encryptor = metaData.findEncryptor(columnIndex);
-        return encryptor.isPresent() ? encryptor.get().decrypt(value.toString()) : value;
+        String ciphertext = (String) mergedResult.getValue(columnIndex, String.class);
+        return null == ciphertext ? null : encryptor.get().decrypt(ciphertext);
```

Comment:
The logic below may better:

- judge queryWithCipherColumn or not
  - if !queryWithCipherColumn, return `mergedResult.getValue(columnIndex, type)`
  - if queryWithCipherColumn,  get value via `mergedResult.getValue(columnIndex, String.class)`
    - if value is null, return null
    - if value is not null, findEncryptor and return decrypt value

Reply:
The encryptor should also judge whether it is null. For example, the id(bigint) is unencrypted and should be executed
`mergedResult.getValue(columnIndex, type)`

Or you mean it's too complicated:
`if (!queryWithCipherColumn || !(encryptor = metaData.findEncryptor(columnIndex)).isPresent())`

Reply:
Yes, it is too complicated

Reply:
Code style fixed